### PR TITLE
[X86] movsd/movss/movd/movq - add support for constant comments

### PR DESCRIPTION
**Pull request by RKSimon**
If we're loading a constant value, print the constant (and the zero upper elements) instead of just the shuffle mask

This did require me to move the shuffle mask handling into addConstantComments as we can't handle this in the MC layer

### CODE REVIEW

File: llvm/test/CodeGen/X86/asm-reg-type-mismatch-avx512.ll [hunk 1]
```diff
@@ -5,7 +5,7 @@ define i64 @test1() nounwind {
 ; CHECK-LABEL: test1:
 ; CHECK:       # %bb.0: # %entry
 ; CHECK-NEXT:    #APP
-; CHECK-NEXT:    vmovq {{.*#+}} xmm16 = mem[0],zero
+; CHECK-NEXT:    vmovq 0, %xmm16
```
Comment (phoebewang) [inline llvm/test/CodeGen/X86/asm-reg-type-mismatch-avx512.ll:8]: I understand the patch just changes the comments, why codegen affected here too?

File: llvm/test/CodeGen/X86/avx512-fma-intrinsics.ll [hunk 2]
```diff
@@ -1150,19 +1150,12 @@ define <16 x float>@test_int_x86_avx512_mask_vfnmadd_ps_512(<16 x float> %x0, <1
...
-; X64-LABEL: foo:
-; X64:       # %bb.0: # %entry
-; X64-NEXT:    vmovss (%rax), %xmm0 # EVEX TO VEX Compression encoding: [0xc5,0xfa,0x10,0x00]
-; X64-NEXT:    # xmm0 = mem[0],zero,zero,zero
-; X64-NEXT:    vfmsub213ss {rd-sae}, %xmm0, %xmm0, %xmm0 # encoding: [0x62,0xf2,0x7d,0x38,0xab,0xc0]
-; X64-NEXT:    retq # encoding: [0xc3]
+; CHECK-LABEL: foo:
+; CHECK:       # %bb.0: # %entry
+; CHECK-NEXT:    vmovss {{.*#+}} xmm0 = mem[0],zero,zero,zero
+; CHECK-NEXT:    # EVEX TO VEX Compression encoding: [0xc5,0xfa,0x10,0x00]
```
Comment (phoebewang) [inline llvm/test/CodeGen/X86/avx512-fma-intrinsics.ll:1156]: The change seems unrelated, need to rebase?

File: llvm/test/CodeGen/X86/asm-reg-type-mismatch-avx512.ll (hunk 1 above)
Comment (RKSimon) [reply llvm/test/CodeGen/X86/asm-reg-type-mismatch-avx512.ll:8]: The instruction comes from inline assembly, when we had the comments in MC we could handle them, but not after moving to MI.

File: llvm/test/CodeGen/X86/avx512-fma-intrinsics.ll (hunk 2 above)
Comment (RKSimon) [reply llvm/test/CodeGen/X86/avx512-fma-intrinsics.ll:1156]: Its an ordering thing - moving to MI from MC means the constant/shuffle comment gets appended before the EVEX pass (see #78585 where I had to clean this up)

**Review by RKSimon — COMMENTED** [review]
A few changes due to the MC-MI diff, there's no clean way to get the ConstantPool data in MC (we already have to do something similar for all the shuffles that load a mask)

File: llvm/test/CodeGen/X86/asm-reg-type-mismatch-avx512.ll (hunk 1 above)
Comment (phoebewang) [reply llvm/test/CodeGen/X86/asm-reg-type-mismatch-avx512.ll:8]: Oh, I misunderstood it. The `0` is memory address, I took it as immediate.

**Review by phoebewang — APPROVED** [review]
LGTM.
